apps/compartmentConverter.cpp

```diff
@@ -234,17 +234,37 @@ int main(const int argc, char** argv)
     float writeTime = clock.getTimef();
 
     const size_t nFrames = (end - start) / step;
+
     boost::progress_display progress(nFrames);
 
-    for (size_t i = 0; i < nFrames; ++i)
+    std::vector<bool> isCompact(gids.size());
+    for (size_t i = 0; i < gids.size(); ++i)
+    {
+        isCompact[i] = _isCompact(in, i);
+    }
+
+    for (size_t frameIndex = 0; frameIndex < nFrames; ++frameIndex)
     {
-        const float t = start + i * step;
+        const float t = start + frameIndex * step;
+
         clock.reset();
-        brion::floatsPtr data = in.loadFrame(t);
+
+        brion::floatsPtr data;
+        try
+        {
+            data = in.loadFrame(t).get();
+        }
+        catch (...)
+        {
+            LBERROR << std::endl
+                    << "Can't load frame at " << t << " ms" << std::endl;
+            ::exit(EXIT_FAILURE);
+        }
         loadTime += clock.getTimef();
         if (!data)
         {
-            LBERROR << "Can't load frame at " << t << " ms" << std::endl;
+            LBERROR << std::endl
+                    << "Can't load frame at " << t << " ms" << std::endl;
```

Comment:
This can go to another pull request where the error handling will be done through exceptions